scripts/create-clog.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env bash
+#
+# create-clog.sh <version> <release-date>
+#
+# Dependencies: sed
+# 
+# Description: Uses template_changelog.md to create a new changelog.
+#
+# Examples:
+#  ./scripts/create-clog.sh "1.18.0" "05/18/2021"
+
+
+set -eou pipefail
+
+VERSION_DELIM="<% X.Y.Z %>"
+RELEASE_DATE_DELIM="<% MM/DD/YYYY %>"
+VERSION=""
+RELEASE_DATE=""
+
+# usage prints information for how to use this program
+usage () {
+  echo "Usage: create-clog <version> <release-date>"
+  echo "Create a changelog from a template"
+  echo "Arugments:"
+  echo "    <version>: A version string 'x.y.z'"
+  echo "    <release-date>: A date string 'mm/dd/yyyy'"
+}
+
+# init parses arguments and initializes all variables.
+init () {
+  lenArgs=2
+  if [ "$#" -ne $lenArgs ]; then
+    echo "illegal number of arguments."
+    echo "Expected: $lenArgs  Received: $#"
+    usage
+    exit 1
+  fi
+
+  if [ -z "${1-}" ]; then
+    echo "version argument is empty."
+    echo "Expected a version string like 'x.y.z'"
+    usage
+    exit 1
+  fi
+  
+  if [ -z "${2-}" ]; then
+    echo "release-date argument is empty."
+    echo "Expected a date string like 'mm/dd/yyyy'"
+    usage
+    exit 1
+  fi
+  
+  VERSION="${1}"
```

Comment:
@deansheather pointed out that `$2` is easier to read, so we should only use `{}` when necessary, this applies to `$VERSION` and `$RELEASE_DATE` too

Reply:
Nice, TIL

Reply:
Do you have a useful summary or reference of when we must use `{}`. We used to always write our scripts that way, so I sort of reach for it reflexively. Again, I'm super happy with the way we're consolidating and standardizing our usage. Thank you!

Reply:
I think `{}` is needed if you're using the default-value syntax `${VERSION:-abc}` to use `$VERSION` if specified, or fall back to `abc` as a default value, or if the text adjacent to your variable contains an underscore or is just letters (e.g. `echo "This is version ${VERSION}_beta"` or `echo "This is version ${VERSION}rc1"`).

In those cases if you're using `set -euo pipefail` as we should be, then you'll just get an error about an undefined variable, so then you know to add the `{}`.